sdk/eventhubs/azure-messaging-eventhubs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,269 @@
+
+An [AmqpException][AmqpException] contains 3 fields which describe the error.
+
+* **getErrorCondition**: The underlying AMQP error. A description of the errors can be found in the [AmqpErrorCondition][AmqpErrorCondition] javadocs or the OASIS AMQP 1.0 spec.
+* **isTransient**: Whether or not trying to perform the same operation is possible.  SDK clients apply the retry policy when the error is transient.
+* **getErrorContext**: Information about where the AMQP error originated.
+  * [LinkErrorContext][LinkErrorContext]: Errors that occur in either the send/receive link.
+  * [SessionErrorContext][SessionErrorContext]: Errors that occur in the session.
+  * [AmqpErrorContext][AmqpErrorContext]: Errors that occur in the connection or a general AMQP error.
+
+### Commonly encountered exceptions
+
+#### amqp:connection:forced and amqp:link:detach-forced
+
+When the connection to Service Bus or Event Hubs is idle, after a period of time, the service will disconnect the client.  This is not a problem as the clients will reestablish a connection with the service. More information: https://docs.microsoft.com/azure/service-bus-messaging/service-bus-amqp-troubleshoot
+
+## Permission issues
+
+An `AmqpException` with an [`AmqpErrorCondition`][AmqpErrorCondition] of "amqp:unauthorized-access" means that the customer's credentials do not allow for them to perform the action (receiving or sending) with Event Hubs.
+
+* [Double check you have the correct connection string][GetConnectionString]
+* [Ensure your SAS token is generated correctly][AuthorizeSAS]
+
+Other possible solutions are listed in [Troubleshoot authentication and authorization issues with Event Hubs][troubleshoot_authentication_authorization]
+
+## Connectivity issues
+
+### Timeout when connecting to service
+
+* Verify that your connection string is correct.
+* Check firewall and ports and that the AMQP port 5671 is open.
+  * Make sure that the endpoint is allowed through firewall.
+* Try using websockets, which connects on port 443. See [Configure web sockets](#configure-web-sockets).
+* See if your network is blocking specific IP addresses.
+  * [What IP addresses do I need to allow?](https://docs.microsoft.com/azure/event-hubs/troubleshooting-guide#what-ip-addresses-do-i-need-to-allow)
+* For more information about troubleshooting network connectivity:
+  * https://docs.microsoft.com/en-us/azure/event-hubs/troubleshooting-guide
+
+### SSL handshake failures
+
+This error can occur when the proxy is not configured properly.  Ask the customer to test their connection to Event Hubs without using a proxy.
+
+### Socket exhaustion errors
+
+Customers should keep an instance of their Event Hub client as long as possible because by default, the creation of a each client results in creating an AMQP connection which uses a socket.  Additionally, clients inherit from `java.io.Closeable`, so they should be calling `close()` when they are finished.
```

Comment:
For clarity, the "client" types are actually EventHubConsumer and EventHubProducer?

(in case someone gets confused since some of our messaging offerings _do_ have a top level client).

Also, I think it's unlikely but there's some connection sharing option in your builder, right? Not sure if that complicates things or not.

Reply:
(you do mention the flag below, but I wasn't sure how that affected your earlier guidance)

Reply:
All of our publicly exposed clients implement Closeable... we'd want them to close anything they create.